pkg/autoupgrade/daemon_test.go

```diff
@@ -0,0 +1,508 @@
+package autoupgrade
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	v1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	kclient "github.com/acorn-io/acorn/pkg/k8sclient"
+	"github.com/acorn-io/baaah/pkg/router"
+	"github.com/acorn-io/baaah/pkg/typed"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type mockClient struct {
```

Comment:
Just a thought, but maybe we could move this to the `testdata` package and rename this to be `mockDaemonClient`?

Reply:
There isn't a `testdata` package here, and this mock is only valuable for unit tests of the `daemonClient`.

I did rename the type to `mockDaemonClient` as you suggested.

Reply:
Yep, totally valid conclusion here.